### PR TITLE
AVRO-2695: avrogen: Non-zero exit code if failure and write error messages to stderr

### PR DESCRIPTION
**Pull request by zcsizmadia**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Avro Jira](https://issues.apache.org/jira/browse/AVRO/) issues and references them in the PR title. For example, "AVRO-1234: My Avro PR"
  - https://issues.apache.org/jira/browse/AVRO-2695
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](https://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

File: lang/csharp/src/apache/codegen/AvroGen.cs [hunk 1]
```diff
@@ -43,9 +43,9 @@ static void Main(string[] args)
                 {
                     if (i + 1 >= args.Length)
                     {
-                        Console.WriteLine("Missing path to protocol file");
+                        Console.Error.WriteLine("Missing path to protocol file");
                         Usage();
-                        return;
+                        return -1;
```
Comment (blachniet) [inline lang/csharp/src/apache/codegen/AvroGen.cs]: A return code of 1 is more common for a general error like this. `avro-tools.jar` returns exits with 1 in these scenarios as well. Let's switch all the `-1` returns to `1`.

File: lang/csharp/src/apache/codegen/AvroGen.cs [hunk 2]
```diff
@@ -23,13 +23,13 @@ namespace Avro
...
     {
-        static void Main(string[] args)
+        static int Main(string[] args)
         {
             // Print usage if no arguments provided or help requested
             if (args.Length == 0 || args[0] == "-h" || args[0] == "--help")
             {
                 Usage();
-                return;
+                return 0;
```
Comment (blachniet) [inline lang/csharp/src/apache/codegen/AvroGen.cs]:
```suggestion
                return 1;
```

File: lang/csharp/src/apache/codegen/AvroGen.cs (hunk 1 above)
Comment (blachniet) [inline lang/csharp/src/apache/codegen/AvroGen.cs]:
```suggestion
                        return 1;
```

File: lang/csharp/src/apache/codegen/AvroGen.cs [hunk 3]
```diff
@@ -55,9 +55,9 @@ static void Main(string[] args)
                 {
                     if (i + 1 >= args.Length)
                     {
-                        Console.WriteLine("Missing path to schema file");
+                        Console.Error.WriteLine("Missing path to schema file");
                         Usage();
-                        return;
+                        return -1;
```
Comment (blachniet) [inline lang/csharp/src/apache/codegen/AvroGen.cs]:
```suggestion
                        return 1;
```

File: lang/csharp/src/apache/codegen/AvroGen.cs [hunk 4]
```diff
@@ -67,17 +67,17 @@ static void Main(string[] args)
                 {
                     if (i + 1 >= args.Length)
                     {
-                        Console.WriteLine("Missing namespace mapping");
+                        Console.Error.WriteLine("Missing namespace mapping");
                         Usage();
-                        return;
+                        return -1;
```
Comment (blachniet) [inline lang/csharp/src/apache/codegen/AvroGen.cs]:
```suggestion
                        return 1;
```

File: lang/csharp/src/apache/codegen/AvroGen.cs [hunk 5]
```diff
@@ -67,17 +67,17 @@ static void Main(string[] args)
...
                     }
 
                     var parts = args[++i].Split(new char[] { ':' }, StringSplitOptions.RemoveEmptyEntries);
                     if (parts.Length != 2)
                     {
-                        Console.WriteLine("Malformed namespace mapping. Required format is \"avro.namespace:csharp.namespace\"");
+                        Console.Error.WriteLine("Malformed namespace mapping. Required format is \"avro.namespace:csharp.namespace\"");
                         Usage();
-                        return;
+                        return -1;
```
Comment (blachniet) [inline lang/csharp/src/apache/codegen/AvroGen.cs]:
```suggestion
                        return 1;
```

File: lang/csharp/src/apache/codegen/AvroGen.cs [hunk 6]
```diff
@@ -88,30 +88,37 @@ static void Main(string[] args)
...
-                Console.WriteLine("Must provide 'outputdir'");
+                Console.Error.WriteLine("Must provide 'outputdir'");
                 isValid = false;
             }
 
+
             if (!isValid)
+            {
                 Usage();
+                rc = -1;
```
Comment (blachniet) [inline lang/csharp/src/apache/codegen/AvroGen.cs]:
```suggestion
                rc = 1;
```

File: lang/csharp/src/apache/codegen/AvroGen.cs [hunk 7]
```diff
@@ -147,10 +154,13 @@ static void Usage()
             }
             catch (Exception ex)
             {
-                Console.WriteLine("Exception occurred. " + ex.Message);
+                Console.Error.WriteLine("Exception occurred. " + ex.Message);
+                return -1;
```
Comment (blachniet) [inline lang/csharp/src/apache/codegen/AvroGen.cs]:
```suggestion
                return 1;
```

File: lang/csharp/src/apache/codegen/AvroGen.cs [hunk 8]
```diff
@@ -169,8 +179,11 @@ static void Usage()
...
```
Comment (blachniet) [inline lang/csharp/src/apache/codegen/AvroGen.cs]:
```suggestion
                return 1;
```

**Comment by blachniet** [comment]
Thanks, @zcsizmadia! This is a good idea. I had one comment about exiting with "1" instead of "-1". After that, we'll get this merged in. Thank you!

**Comment by Fokko** [comment]
Restarted the CI

**Pull request merged**